### PR TITLE
Updated Terraform readme: Alex

### LINKED ISSUE #200: Update Terraform README.md
As a developer, I want clear documentation that outlines the installation process, the structure of the pipeline, and how to run the app, so new contributors can easily get started and understand the functionality of the app.

- Edit README for terraform so that it is up to date for the project

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #200 

What I have done:
- Updated terraform readme
- Included information on `main.tf`

### CODE REVIEW

File: terraform/README.md [hunk 1]
```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
 ## How it works
 #### main.tf
 - This contains the terraform config required for provisioning the cloud architecture.
```
Comment (MurasatoNaoya) [inline terraform/README.md:62]: Clear explanation of what exactly is being built, and a high-level overview of how exactly the main.tf file does this; like for the iterative generation of Lambdas for example.

File: terraform/README.md [hunk 2]
```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
...
+    - ECS:
+        - IAM role - an administration role that allows the ECS task to execute.
+        - AWS cloudwatch rules - allows for logging by ecs.
+        - ECS task definition - this defines which ECR image will be loaded and on what machine.
+        - ECS service - a system that restarts the ECS task upon failure.
+    - Eventbridge:
+        - Cloudwatch roles - allows for logging by eventbridge
+        - Eventbridge bus - A quadhoral, hourly and weekly eventbridge triggers are provisioned
+
 #### variables.tf
```
Comment (MurasatoNaoya) [inline terraform/README.md:92]: tf.vars of course in the environment variables / .env equivalent for Terraform, the main.tf file needs these variables, so it's good to mention for someone unfamiliar with Terraform.

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

File: terraform/README.md [hunk 3]
```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
 ## How it works
```
[the remark this answers is not in the document]
Comment (CalebMathers) [reply terraform/README.md:60]: I agree, the use of markdown is good, handsdown

File: terraform/README.md [hunk 4]
```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
...
 - This contains the terraform config required for provisioning the cloud architecture.
+- Existing resources are specified using the `data` keyword they include:
+    - VPC - Virtual private cloud.
+    - Public Subnet - a subsection of the VPC that is accessible externally.
+    - ECS cluster - this is the cluster of machines our dashboard will run on.
+- Local variables are defined by the `locals` keyword this defines:
+    - API keys - for running the lambda pipelines.
+    - Database variables - to securely access the RDS instance.
+    - Image URIs - for the ECR repositories to be used.
+- New resources are defined by the `resource` keyword and these services can be grouped into the following:
```
Comment (CalebMathers) [inline terraform/README.md:71]: Really useful to explain all the resources inside

File: terraform/README.md [hunk 5]
```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
...
+        - AWS cloudwatch rules - allows for logging by ecs.
+        - ECS task definition - this defines which ECR image will be loaded and on what machine.
+        - ECS service - a system that restarts the ECS task upon failure.
+    - Eventbridge:
+        - Cloudwatch roles - allows for logging by eventbridge
+        - Eventbridge bus - A quadhoral, hourly and weekly eventbridge triggers are provisioned
+
 #### variables.tf
 - Defines the secret variables used in `main.tf`.
+- Contains no sensitive information, only the name and type of each sensitive variable
```
Comment (CalebMathers) [inline terraform/README.md:94]: Important to make this clear; nice

File: terraform/README.md
Comment (CalebMathers) [inline terraform/README.md:1]: LGTM

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**